public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.tsx

```diff
@@ -119,7 +119,7 @@ export function InstallControlsButton({
 
   let disableUninstall =
     config.pluginAdminExternalManageEnabled && configCore.featureToggles.managedPluginsInstall
-      ? plugin.isUninstallingFromInstance
+      ? plugin.isUninstallingFromInstance || !plugin.isFullyInstalled || plugin.isUpdatingFromInstance
```

Comment:
Is it possible to create a function and name it with logic behind `plugin.isUninstallingFromInstance || !plugin.isFullyInstalled || plugin.isUpdatingFromInstance` check? Encapsulate Predicate refactor.